tools/oas-to-am/bin/oas-to-am.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "Licens
+# you may not use this file except in compliance with the Lic
+# You may obtain a copy of the License at
+#
+# <http://www.apache.org/licenses/LICENSE-2.0>
+#
+# Unless required by applicable law or agreed to in writing,
+# distributed under the License is distributed on an "AS IS"
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expres
+# See the License for the specific language governing permiss
+# limitations under the License.
+
+# shellcheck disable=2140
+
+set -e
+
+[ -z "$SPEC" ] && echo "Specification Missing" && exit
+[ -z "$OPERATION" ] && echo "Operation Missing" && exit
+
+# Build up AssignMessage policy
+RESULT="<AssignMessage name="Assign.$OPERATION">"
+RESULT="$RESULT<AssignTo createNew="true" type="request">$OPERATION</AssignTo>"
+
+# Set target url as a variable for use in TargetEndpoint, ServiceCallout or TargetServer
+RESULT="$RESULT<AssignVariable>"
+RESULT="$RESULT<Name>$OPERATION.url</Name>"
```

Comment:
Just an observation: The input variables are generic as `custom.*` but the output is specific to the operation. Obviously depending on the use case either one could be useful. In a scenario where you'd have conditional flows and put these generated AssignMessages policies in, that would be easier to just reuse the same variable name for all request URLs.

Reply:
true - will put as custom for now